surveyor/collector_statz.go

```diff
@@ -638,31 +639,53 @@ type connzAggregate struct {
 	msgsRecv  float64
 }
 
-func getConnzAggregate(nc *nats.Conn, account string) (connzAggregate, error) {
+func getConnzAggregate(nc *nats.Conn, numServers int, account string) (connzAggregate, error) {
 	// TODO: Replace with "$SYS.REQ.ACCOUNT.%s.CONNS" after NATS 2.8.4.
 	// CONNS returns bytes sent/recv at the account level without needing the
 	// following code.
 	subj := fmt.Sprintf("$SYS.REQ.ACCOUNT.%s.CONNZ", account)
-	msg, err := nc.Request(subj, nil, 3*time.Second)
+
+	rep := nc.NewRespInbox()
+
+	msg := nats.NewMsg(subj)
+	msg.Reply = rep
+	msg.Data = nil
+
+	s, err := nc.SubscribeSync(msg.Reply)
 	if err != nil {
 		return connzAggregate{}, err
 	}
 
-	var r server.ServerAPIResponse
-	var d server.Connz
-	r.Data = &d
-	if err := json.Unmarshal(msg.Data, &r); err != nil {
+	if err := nc.PublishMsg(msg); err != nil {
 		return connzAggregate{}, err
 	}
 
 	var agg connzAggregate
-	for _, c := range d.Conns {
-		agg.bytesSent += float64(c.InBytes)
-		agg.bytesRecv += float64(c.OutBytes)
+	var r server.ServerAPIResponse
+	var d server.Connz
+	r.Data = &d
 
-		agg.msgsSent += float64(c.InMsgs)
-		agg.msgsRecv += float64(c.OutMsgs)
+	for i := 0; i < numServers; i++ {
+		m, err := s.NextMsg(3 * time.Second)
```

Comment:
This should probably be the `PollTimeout` from the Surveyor `Options` struct?

Reply:
I just kept what was there. It's 3 seconds everywhere else. We can change it in all the other account functions if that's what it's supposed to be.

Reply:
Could either fix here or in follow-up PR though, if it's a bigger scoped change

Reply:
yeah I"m in favor of a follow up because these functions will most likely become methods on the statzcollector. I don't want to pass just an argument for the time.